src/routes/blog/+page.svelte

```diff
@@ -1,87 +1,47 @@
 <script>
-	import { SITE_TITLE } from '$lib/siteConfig';
+	import { browser } from '$app/env';
+	import { goto } from '$app/navigation';
+	import { page } from '$app/stores';
 
-	import queryString from 'query-string';
-	import { onMount } from 'svelte';
+	import { SITE_TITLE, POST_CATEGORIES } from '$lib/siteConfig';
 
 	import IndexCard from '../../components/IndexCard.svelte';
 
-
 	/** @type {import('./$types').PageData} */
 	export let data;
 
 	// technically this is a slighlty different type because doesnt have 'content' but we'll let it slide
 	/** @type {import('$lib/types').ContentItem[]} */
 	$: items = data.items; 
-	
-
-	let inputEl;
-	let search;
 
-	let urlState = { filter: '', show: [] };
-	let defaultURLState = { filter: '', show: [] };
-
-	const setURLState = (newState) => {
-		const finalState = { ...urlState, ...newState }; // merge with existing urlstate
-		urlState = finalState;
-		Object.keys(finalState).forEach(function (k) {
-			if (
-				// don't save some state values if it meets the conditions below
-				!finalState[k] || // falsy
-				finalState[k] === '' || // string
-				(Array.isArray(finalState[k]) && !finalState[k].length) || // array
-				finalState[k] === defaultURLState[k] // same as default state, unnecessary
-			) {
-				delete finalState[k]; // drop query params with new values = falsy
-			}
-		});
-		if (typeof window !== 'undefined')
-			history.pushState(
-				{},
-				'',
-				document.location.origin +
-					document.location.pathname +
-					'?' +
-					queryString.stringify(finalState)
-			);
-	};
+	function searchParamToArray(key) {
+		return ($page.url.searchParams.get(key) || '').split(',').filter((e) => e);
+	}
 
-	onMount(() => {
-		if (location.search.length < 1) return; // early terminate if no search
-		let givenstate = queryString.parse(location.search);
-		if (!Array.isArray(givenstate.show)) givenstate.show = [givenstate.show];
-		// if (!givenstate.show.includes('Essays')) essays = false;
-		// if (!givenstate.show.includes('Talks')) talks = false;
-		// if (!givenstate.show.includes('Podcasts')) podcasts = false;
-		// if (!givenstate.show.includes('Tutorials')) tutorials = false;
-		// if (!givenstate.show.includes('Snippets')) snippets = false;
-		// if (!givenstate.show.includes('Notes')) notes = false;
-		if (givenstate.filter) search = givenstate.filter;
-		urlState = { ...defaultURLState, ...givenstate };
-	});
-	function saveURLState() {
-		setTimeout(() => {
-			setURLState({
-				filter: search,
-				show: [
-					// essays && 'Essays',
-					// talks && 'Talks',
-					// podcasts && 'Podcasts',
-					// snippets && 'Snippets',
-					// tutorials && 'Tutorials',
-					// notes && 'Notes'
-				].filter(Boolean)
-			});
-		}, 100);
```

Comment:
this code is required for URLs to be copy pastable

eg when i search something on my site it shows up in the URL, and can be copy pasted

![image](https://user-images.githubusercontent.com/6764957/185957103-23bb7635-0e82-4a81-8fea-8d4a04999bf7.png)


but on this PR it doesnt

![image](https://user-images.githubusercontent.com/6764957/185957028-1dd8e4a1-a287-4d56-8507-6cafdc1776c6.png)

Reply:
[14691a7](https://github.com/sw-yx/swyxkit/pull/97/commits/14691a773585345df798423901576f72ecf016a8)
Fixed the URLs to be updated with search filter.